nominal/core/unit.py

```diff
@@ -0,0 +1,66 @@
+import dataclasses
+
+from typing_extensions import Self
+
+from .._api.combined import datasource_api, scout_asset_api, scout_run_api, scout_units_api
+
+
+@dataclasses.dataclass(frozen=True)
+class Unit:
+    """Plain python wrapper to encapsulate the data from scout_units_api.Unit.
+
+    This is primarily used when setting or retrieving the units of a channel within a dataset.
+    """
+
+    # Plain english name of the unit (e.g. 'coulomb')
+    name: str
+
+    # Abbreviated symbol for the unit (e.g. 'C')
+    # See: https://ucum.org/ucum
+    symbol: str
+
+    # General physical property being measured by the unit (e.g. 'electric charge')
+    # Units may only be converted to one another if they measure the same property
+    property: str | None
+
+    # Physical dimensions in terms of the base units of the system.
+    # Units may only be converted to one another if they have the same dimension.
+    dimension: dict[str, int] | None
+
+    # System of measurement containing this unit.
+    # Currently, all units come from the Universal Code for Units of Measure (UCUM).
+    system: str | None
+
+    @classmethod
+    def _from_units_api_unit(cls, api_unit: scout_units_api.Unit) -> Self:
+        """Construct a Unit from a fully-fleshed-out conjure Unit."""
+
+        if api_unit.name is None:
+            raise RuntimeError(f"Cannot deserialize api unit-- missing name: {api_unit}")
+        elif api_unit.symbol is None:
+            raise RuntimeError(f"Cannot deserialize api unit-- missing symbol: {api_unit}")
+
+        base_dimensions = api_unit.dimension.base_dimensions if api_unit.dimension is not None else None
+        return cls(
+            name=api_unit.name,
+            symbol=api_unit.symbol,
+            property=api_unit.property,
+            dimension=base_dimensions,
+            system=api_unit.system,
+        )
+
+    @classmethod
+    def _from_simple_api_unit(cls, api_unit: scout_run_api.Unit | scout_asset_api.Unit | datasource_api.Unit) -> Self:
```

Comment:
hmm a little unsure about having these properties be exposed if their semantic meaning can be lost via being constructed differently

Reply:
maybe lets only keep the basic attributes exposed for now, until we can get the backend to always return the enriched unit types?

Reply:
alternatively, if everyone is agreed on making the backend change, we can always make the extra network round trip to get all the unit information and stitch it together here, and then in the future when we do change the backend impl, the client impl can change without any difference to users